rest-api/src/main/scala/za/co/absa/enceladus/rest_api/controllers/v3/ControllerPagination.scala

```diff
@@ -74,12 +74,14 @@ object ControllerPagination {
    * @param defaultValue value to use if extraction fails
    * @return On extraction success, `extractedIntValue` is returned, otherwise (empty or invalid) `defaultValue` is returned.
    */
-  private def extractDefinedValueOrDefault(optField: Option[String], defaultValue: Int): Int = {
+  private def extractDefinedValueOrDefault(optField: Option[String], defaultValue: Int, paramNameHint: String): Int = {
     optField match {
       case None => defaultValue
       case Some(intAsString) => Try(intAsString.toInt) match {
-        case Success(value) => value
-        case Failure(_) => defaultValue
+        case Success(value) if value >= 0 => value
+        case Success(value) => throw new IllegalArgumentException(s"Value '$value' must be > 0 for the '$paramNameHint' param.")
+        case Failure(_) =>
+          throw new IllegalArgumentException(s"'$intAsString' could not be interpreted as int for the '$paramNameHint' param.")
```

Comment:
Can you please help me understand how this exception gets translated into 400?

Reply:
Certainly.

In "normal" scala/java code, it is passed along as a regular exception, with stack trace and all, but for Spring's REST, we define a [RestExcepstionHandler](https://github.com/AbsaOSS/enceladus/blob/develop/rest-api/src/main/scala/za/co/absa/enceladus/rest_api/controllers/RestExceptionHandler.scala#L47) that transforms known exceptions into specific REST responses (unmatched ones are passed as 500s with a stack trace).

Since our handler is defined for the `IllegalArgumentException` type, we can instill this behavior in integTests, too.